app/services/registration_checker.rb

```diff
@@ -200,5 +216,26 @@ def existing_registration_in_series?
       end
       false
     end
+
+    def competitor_qualifies_for_event?(event, qualification)
+      competitor_qualification_results = UserApi.qualifications(@requestee_user_id)
+      result_type = qualification['resultType']
+
+      competitor_pr = competitor_qualification_results.find { |result| result['eventId'] == event && result['type'] == result_type }
+      return false if competitor_pr.blank?
```

Comment:
I'm explicitly returning booleans here which feels like a huge code smell, but I haven't been able to come up with something better.

One alternative might be to change this to a `!` method and raise errors instead of returning a boolean?

Reply:
Well, since boolean is the expected return type of this method, I don't see what's wrong with it.